### PR TITLE
Decouple display and gamestate.

### PR DESCRIPTION
**Pull request by kfwong**
Added hooks into the webrtc display component so the gamestate can react to the event. Parameters (playerId, data) are passed into the function by `SumoDisplay` so other class can make use of it.

Usage:
```javascript
var display = new SumoDisplay();
display.onPlayerCreated = playerId => { ... }
display.onPlayerDisconnected = playerId => { ... }
display.onPlayerData = data => { ... }
display.onRoomCreatedSuccess = roomKey => { ... }
display.onRoomCreatedFail = roomKey => { ... }

display.start(roomKey);
```

### CODE REVIEW

File: WebRTCTemplate/display.js [hunk 1]
```diff
@@ -30,17 +27,28 @@ class SumoDisplay {
...
+
+        this.onPlayerCreated = new Function();
+        this.onPlaterDisconnected = new Function();
+        this.onData = new Function();
+    }
+
+    isRoomExists(roomKey){
+        console.log(`Checking if room "${roomkey} exists."`)
+
+        return this.db.collection('rooms').doc(coomKey).get();
```
Comment (nicholasluimy) [inline WebRTCTemplate/display.js]:
```suggestion
        return this.db.collection('rooms').doc(roomKey).get();
```

File: WebRTCTemplate/display.js [hunk 2]
```diff
@@ -30,17 +27,28 @@ class SumoDisplay {
...
 
-        this.rooms = this.db.collection('rooms');
-
         // Initialize a hashmap to keep track of connected players
         this.players = {};
 
-        this.gameState = new GameState();
-        this.roomKey = "default";
+        // TODO
+        //this.gameState = new GameState();
```
Comment (nicholasluimy) [inline WebRTCTemplate/display.js]: Just to clarify, this means that GameState is a separate singleton, that we declare and manage in host-logic?

Meaning that, host-logic is the host object, that separately manages a network component (`SumoDisplay`) and a game state layer (`GameState`) over unity

Comment (kfwong) [reply WebRTCTemplate/display.js]: Yes correct. It's better than having a linear dependecies. Prefer to have `SumoDisplay` and `SumoClient` to be open for extension but close for modification.

Comment (kfwong) [reply WebRTCTemplate/display.js]: I also think that GameState and host-logic can be combined uh. Less files to cross reference maybe hahaha.

**Pull request merged**